src/visualization.jl

```diff
@@ -0,0 +1,617 @@
+    plt = plot(mesh;plot_params...)
+    WriteVTK.vtk_grid(f,filename,plt;vtk_grid_params...)
+end
+
+function WriteVTK.vtk_grid(f::Function,filename,dom::AbstractDomain;plot_params=(;),vtk_grid_params...)
+    plt = plot(dom;plot_params...)
+    WriteVTK.vtk_grid(f,filename,plt;vtk_grid_params...)
+end
+
+struct PVD{A} <: AbstractType
+    pvd::A
+end
+
+struct PPVD{A} <: AbstractType
+    partition::A
+end
+
+function WriteVTK.close(pvd::PVD)
+    WriteVTK.close(pvd.pvd)
+end
+
+function WriteVTK.close(ppvd::PPVD)
+    map_main(WriteVTK.close,ppvd.partition)
+end
+
+function Base.setindex!(a::PVD,plt::VTKPlot,time)
+    a.pvd[time] = plt.vtk
+end
+
+function Base.setindex!(a::PPVD,plt::VTKPlot,time)
+    map_main(a.partition,plt.vtk) do a,vtk
+        a[time] = vtk
+    end
+end
+
+function WriteVTK.paraview_collection(filename,pplt::Plot;kwargs...)
+    WriteVTK.paraview_collection(filename;kwargs...) |> PVD
+end
+
+function WriteVTK.paraview_collection(filename,pplt::PPlot;kwargs...)
+    pvds = map_main(partition(pplt)) do _
+        WriteVTK.paraview_collection(filename;kwargs...)
+    end
+    PPVD(pvds)
+end
+
+function WriteVTK.paraview_collection(f::Function,filename,plt::Union{Plot,PPlot};kwargs...)
+    vtk = WriteVTK.paraview_collection(filename,plt;kwargs...)
+    files = nothing
+    try
+        f(vtk)
+    finally
+        files = close(vtk)
+    end
+    files
+end
+
+function WriteVTK.paraview_collection(filename,mesh::Union{AbstractMesh,PMesh};plot_params=(;),vtk_grid_params...)
+    plt = plot(mesh;plot_params...)
+    WriteVTK.paraview_collection(filename,plt;vtk_grid_params...)
+end
+
+function WriteVTK.paraview_collection(filename,dom::AbstractDomain;plot_params=(;),vtk_grid_params...)
+    plt = plot(dom;plot_params...)
+    WriteVTK.paraview_collection(filename,plt;vtk_grid_params...)
+end
+
+function WriteVTK.paraview_collection(f::Function,filename,mesh::Union{AbstractMesh,PMesh};plot_params=(;),vtk_grid_params...)
+    plt = plot(mesh;plot_params...)
+    WriteVTK.paraview_collection(f,filename,plt;vtk_grid_params...)
+end
+
+function WriteVTK.paraview_collection(f::Function,filename,dom::AbstractDomain;plot_params=(;),vtk_grid_params...)
+    plt = plot(dom;plot_params...)
+    WriteVTK.paraview_collection(f,filename,plt;vtk_grid_params...)
+end
+
+# Makie prototype functions to be defined inside Makie's extension module; see ext/GalerkinToolkitMakieExt.jl
+
+function makieplot end
+function makieplot! end
+function makie0d end
+function makie0d! end
+function makie1d end
+function makie1d! end
+function makie2d end
+function makie2d! end
+function makie2d1d end
+function makie2d1d! end
+function makie3d end
+function makie3d! end
+function makie3d1d end
+function makie3d1d! end
+
+# handler for displaying a hint in case the user tries to call functions defined in the extension module
+# https://github.com/JuliaLang/julia/blob/b9d9b69165493f6fc03870d975be05c67f14a30b/base/errorshow.jl#L1039
+
+function __init__()
```

Comment:
I am not sure about this __init__ hook. Please comment it out for the moment.

Reply:
Yes. This is added to issue a message to the user in case he/she calls, e.g., `GT.makie1d(...)`, but forgets to import Makie before hand. The drawback here is that this is still experimental (`Base.Experimental`) and adds more core. This is up to you to keep it (in the main modules as suggested) or comment out for the moment.

Reply:
We comment it out for the moment.